### ISSUE TITLE
Problem with mongo+srv dsn, when DNS SRV response contains more that 4 items

### ISSUE THREAD

**Issue by vsychov** [mongodb/mongo-php-driver#1091]
### Bug Report

Hi, I found problem with `mongodb+srv` dsn. When SRV response, contains 4 or less items, everything works great, but if it contains 5 or more items, driver is crash.

Example of SRV response (successful example):
```
root@php-cli:/code# nslookup -type=SRV _mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local
Server:		10.96.0.10
Address:	10.96.0.10#53

_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 25 27017 my-cluster-name-rs0-1.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 25 27017 my-cluster-name-rs0-arbiter-0.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 25 27017 my-cluster-name-rs0-0.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 25 27017 my-cluster-name-rs0-2.my-cluster-name-rs0.default.svc.cluster.local.
```

and result of script execution as expected, and return cursor:
```
object(MongoDB\Driver\Cursor)#4 (10) {
  ["database"]=>
  string(4) "test"
  ...
```

but when I add new item to SRV response:
```
root@php-cli:/code# nslookup -type=SRV _mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local
;; Truncated, retrying in TCP mode.
Server:		10.96.0.10
Address:	10.96.0.10#53

_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 20 27017 my-cluster-name-rs0-1.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 20 27017 my-cluster-name-rs0-arbiter-0.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 20 27017 my-cluster-name-rs0-arbiter-1.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 20 27017 my-cluster-name-rs0-0.my-cluster-name-rs0.default.svc.cluster.local.
_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local	service = 0 20 27017 my-cluster-name-rs0-2.my-cluster-name-rs0.default.svc.cluster.local.
```

Then php script crash:
```
root@php-cli:/code# php test.php

Fatal error: Uncaught MongoDB\Driver\Exception\ConnectionTimeoutException: Invalid SRV answer for "_mongodb._tcp.my-cluster-name-rs0.default.svc.cluster.local" in /code/test.php:4
Stack trace:
#0 /code/test.php(4): MongoDB\Driver\Manager->executeCommand('test', Object(MongoDB\Driver\Command))
#1 {main}
  thrown in /code/test.php on line 4
```

### Environment

official php docker image, with installed `mongodb` driver via `pecl`, with ssl support.
```
root@php-cli:/code# php -v
PHP 7.4.2 (cli) (built: Feb  1 2020 19:33:30) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
```

```
root@php-cli:/code#   php -i | grep -E 'mongodb|libmongoc|libbson'
Additional .ini files parsed => /usr/local/etc/php/conf.d/docker-php-ext-mongodb.ini,
mongodb
libbson bundled version => 1.15.2
libmongoc bundled version => 1.15.2
libmongoc SSL => enabled
libmongoc SSL library => OpenSSL
libmongoc crypto => enabled
libmongoc crypto library => libcrypto
libmongoc crypto system profile => disabled
libmongoc SASL => disabled
libmongoc ICU => disabled
libmongoc compression => enabled
libmongoc compression snappy => disabled
libmongoc compression zlib => enabled
mongodb.debug => no value => no value
```

### Test Script

```
<?php
$mongoDsn = 'mongodb+srv://user:password@my-cluster-name-rs0.default.svc.cluster.local/admin?replicaSet=rs0&ssl=false';
$manager = new MongoDB\Driver\Manager($mongoDsn);
var_dump($manager->executeCommand('test', new MongoDB\Driver\Command(['ping' => 1])));
```

**Comment by alcaeus** [comment]
The issue you report sounds similar to an issue reported in libmongoc: https://jira.mongodb.org/browse/CDRIVER-3388. In that case, the user saw similar symptoms where adding an additional SRV record caused connection errors.

This bug was fixed in libmongoc 1.16.0, which will be shipped with the upcoming 1.7.0 driver version. This release is scheduled to happen later today.

**Comment by alcaeus** [comment]
1.7.0 was released a while back, but I forgot to close this issue. Please report back if the problem has not been solved for you.
